joomla/joomla-cms
# [4.0] #__workflows DEFAULT NOW() for created and modified leads to database error

Pull Request for Issues https://github.com/joomla/joomla-cms/issues/21574 (part "No user after installation" and "Workflow missing tables" https://github.com/joomla/joomla-cms/issues/21582 .

### Summary of Changes
- Change default `NOW()` of columns `modified` and `created` to `0000-00-00 00:00:00` (like common in Joomla)

### Testing Instructions
I don't know exactly if it's a MySql version issue or not.

- Provider All-Inkl.
- PHP version 7.1.19-nmm1 and 7.2.7
- **MySql 5.5.58**-nmm1-log

- Install nightly build FULL package of 2018-08-12 or 2018-08-13 where com_workflow is already merged.
- During installation don't check option to delete /installation/ folder.
- After installation check database tables #__users, #__workflow, #__workflow_stages, #__workflow_transitions.
- They are empty or missing but shouldn't.
- You cannot login in backend.
- Delete file configuration.php.
- Change file  installation/sql/mysql/joomla.sql like done in this PR.
- Start a new installation by reloading the Joomla page.
- Check again.
- All fine.

### Expected result
- Complete installation.
- Empty or no /administrator/logs/error.php after installation.
- No Congratulations message after install(!!) Not part of this pr!

### Actual result
`"Invalid default value for 'created'"`

error.php shows the following stack trace after installation
```
CRITICAL ... error	Uncaught Throwable of type 

Joomla\Database\Exception\PrepareStatementFailureException thrown with message "Invalid default value for 'created'".

Stack trace:

#0 .../libraries/vendor/joomla/database/src/Mysqli/MysqliDriver.php(873): Joomla\Database\Mysqli\MysqliStatement->__construct(Object(mysqli), 'CREATE TABLE IF...')

#1 .../libraries/vendor/joomla/database/src/DatabaseDriver.php(1778): Joomla\Database\Mysqli\MysqliDriver->prepareStatement('CREATE TABLE IF...')

#2 .../installation/src/Model/DatabaseModel.php(1126): Joomla\Database\DatabaseDriver->setQuery(Object(Joomla\Database\Mysqli\MysqliQuery))

#3 .../installation/src/Model/DatabaseModel.php(565): Joomla\CMS\Installation\Model\DatabaseModel->populateDatabase(Object(Joomla\Database\Mysqli\MysqliDriver), 'sql/mysql/jooml...')

#4 .../installation/src/Model/ConfigurationModel.php(71): Joomla\CMS\Installation\Model\DatabaseModel->createTables(Object(stdClass))

#5 .../installation/src/Controller/InstallationController.php(95): Joomla\CMS\Installation\Model\ConfigurationModel->setup(Object(stdClass))

#6 .../libraries/src/MVC/Controller/BaseController.php(717): Joomla\CMS\Installation\Controller\InstallationController->config()

#7 .../installation/src/Application/InstallationApplication.php(316): Joomla\CMS\MVC\Controller\BaseController->execute('config')

#8 .../installation/src/Application/InstallationApplication.php(202): Joomla\CMS\Installation\Application\InstallationApplication->executeController()

#9 .../installation/src/Application/InstallationApplication.php(231): Joomla\CMS\Installation\Application\InstallationApplication->dispatch()

#10 .../installation/src/Application/InstallationApplication.php(246): Joomla\CMS\Installation\Application\InstallationApplication->doExecute()

#11 .../installation/includes/app.php(65): Joomla\CMS\Installation\Application\InstallationApplication->execute()

#12 .../installation/index.php(36): require_once('/www/htdocs/w00...')

#13 {main}
```

This means DatabaseModel::populateDatabase() dies silently 

### Additional comment
- No Congratulations message after install(!!) Not part of this pr!
